vacationtime-solution/src/Code.js

```diff
@@ -0,0 +1,186 @@
+        COLUMN_NUMBER.EMAIL];
+    var managerCal = CalendarApp.getCalendarById(managerEmail);
+    var employeeEmail = rangeValues[i][0];
+    var employeeName = rangeValues[i][1];
+    var startDate = rangeValues[i][4];
+    var endDate = rangeValues[i][5];
+    var approvalStatus = rangeValues[i][6];
+                                        
+    // Creates a calendar event.
+    var event = managerCal.createEvent('APPROVED VACATION TIME FOR ' +
+        employeeName, startDate, endDate, {
+        description: 'Your vacation time from ' +
+                                            startDate + ' to ' + endDate +
+                                            ' has been approved! Enjoy!',
+        guests: employeeEmail,
+        sendInvites: true,
+    });
+}
+
+/**
+* Checks the approval status of each employee and notifies
+* them of their status accordingly.
+**/
+function checkApproval(rangeValues, i, currentStartRow) {
+    var sheet = SpreadsheetApp.getActiveSheet();
+    var rangeValues = rangeValues;
+    var currentStartRow = currentStartRow;
+    var i = i;
+    
+    // Gets necessary values.
+    var employeeEmail = rangeValues[i][0];
+    var employeeName = rangeValues[i][1];
+    var approvalStatus = rangeValues[i][6];
+    
+    // Checks approval status.
+    if (approvalStatus == 'NOT APPROVED') {
+      EMAIL_SUBJECT = 'ERR: Vacation Time NOT Approved';
+      EMAIL_MESSAGE = 'time not approved';
+        
+      // Sends email of disapproval.
+      MailApp.sendEmail(employeeEmail, EMAIL_SUBJECT, EMAIL_MESSAGE);
+      sheet.getRange(currentStartRow, COLUMN_NUMBER.NOTIFIED)
+          .setValue('NOTIFIED');
+    } else if (approvalStatus == 'APPROVED') {
+        createCalEvent(rangeValues, i);
+        sheet.getRange(currentStartRow, COLUMN_NUMBER.NOTIFIED)
+            .setValue('NOTIFIED');
+    }
+}
+
+/**
+* Checks the notification status of each employee and, if not notified,
+* notifies them of their status accordingly, through use of helper 
+* functions.
+*/
+function notifyEmployees() {
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastRow = sheet.getLastRow();
+  var lastCol = sheet.getLastColumn();
+  var frozenRows = sheet.getFrozenRows();
+  var beginningRow = frozenRows + 1;
+  var numRows = lastRow - frozenRows;
+
+  // Creates notified column.
+  notifiedColRange = createNotifiedColumn();
+  var notifiedStatus = notifiedColRange.getValues();
+
+  for (var i = 0; i <= lastRow - beginningRow; i++) {
+    // Ensures does not notify twice.
+    if (notifiedStatus[i][0] == 'NOTIFIED') {
+      continue;
+    }
+    
+    var currentStartRow = i + beginningRow;
+    
+    // Obtains necessary values.
+    var rangeValues = sheet.getRange(currentStartRow, COLUMN_NUMBER.EMAIL,
+        1, COLUMN_NUMBER.APPROVAL - COLUMN_NUMBER.EMAIL + 1).getValues();
+      
+    // Checks approval of each employee & notifies them accordingly.
+    checkApproval(rangeValues, i, currentStartRow);
```

Comment:
Why don't you review the status in this method and use a smaller helper function to send the emails